test/e2e/multicast_test.go

```diff
@@ -158,7 +278,261 @@ type multicastTestcase struct {
+		for _, senderConfig := range mc.senderConfigs {
+			stats := mc.antctlResults[senderConfig.name]
+			t.Logf("check antctl get podmulticaststats result for %s", senderConfig.name)
+			antreaPod, err := data.getAntreaPodOnNode(senderConfig.nodeName)
+			if err != nil {
+				t.Errorf("error get getAntreaPodOnNode for %s", senderConfig.name)
+				failOnError(err, t)
+			}
+			isMatch, err := checkAntctlResult(t, data, antreaPod, senderConfig.name, stats.Inbound, stats.Outbound)
+			if err != nil || !isMatch {
+				return false, err
+			}
+		}
+		receiversAddresses := sets.NewString()
+		for _, receiverConfig := range mc.receiverConfigs {
+			receiversAddresses.Insert(receiverConfig.IPs...)
+			stats := mc.antctlResults[receiverConfig.name]
+			t.Logf("check antctl get podmulticaststats result for %s", receiverConfig.name)
+			antreaPod, err := data.getAntreaPodOnNode(receiverConfig.nodeName)
+			if err != nil {
+				t.Errorf("error get getAntreaPodOnNode for %s", receiverConfig.name)
+				failOnError(err, t)
+			}
+			isMatch, err := checkAntctlResult(t, data, antreaPod, receiverConfig.name, stats.Inbound, stats.Outbound)
+			if err != nil || !isMatch {
+				return false, err
+			}
+		}
+		for _, anp := range mc.igmpANPConfigs {
+			stats, _ := data.crdClient.StatsV1alpha1().AntreaNetworkPolicyStats(data.testNamespace).Get(context.TODO(), anp.name, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			t.Logf("Got AntreaNetworkPolicy stats: %v", stats)
+			expectedRuleNames, exist := mc.igmpStatsResult[anp.name]
+			if !exist || len(expectedRuleNames) == 0 {
+				if len(stats.RuleTrafficStats) > 0 {
+					return false, nil
+				}
+			} else {
+				ruleNames := sets.NewString()
+				for _, ruleName := range stats.RuleTrafficStats {
+					ruleNames.Insert(ruleName.Name)
+				}
+				if !ruleNames.Equal(expectedRuleNames) {
+					return false, nil
+				}
+			}
+		}
+		for _, anp := range mc.multicastANPConfigs {
+			stats, err := data.crdClient.StatsV1alpha1().AntreaNetworkPolicyStats(data.testNamespace).Get(context.TODO(), anp.name, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			t.Logf("Got AntreaNetworkPolicy stats: %v", stats)
+			if len(stats.RuleTrafficStats) != len(mc.multicastANPStatsResult[anp.name]) {
+				return false, nil
+			}
+			for _, rule := range stats.RuleTrafficStats {
+				pktCount := mc.multicastANPStatsResult[anp.name][rule.Name]
+				if pktCount != rule.TrafficStats.Packets {
+					return false, nil
+				}
+			}
+		}
+		for _, address := range receiversAddresses.List() {
+			multicastGroup, err := data.crdClient.StatsV1alpha1().MulticastGroups().Get(context.TODO(), address, metav1.GetOptions{})
+			t.Logf("Got multicast group information for group %s: %v", address, multicastGroup)
+			t.Logf("Got multicastGroup error %v", err)
+			expectedpodNames, exist := mc.multicastGroupsResult[address]
+			if !exist || len(expectedpodNames) == 0 {
+				if err == nil {
```

Comment:
So if err != nil, it will finally return true?

Reply:
updated